### PR TITLE
FCFS algorithm in C

### PR DESCRIPTION
**Pull request by Bhumika2409**
1. Created first_come_first_serve_algorithm.c file in operating_system folder of C.
2. updated the C/README.md file for the ame

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
FCFS is a very famous scheduling algorithm that is used in operating systems and I have written the C code for the same

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
 #6790

### CODE REVIEW

**Comment by Bhumika2409** [comment]
@HarshCasper @ankitaggarwal23 @bhav09 @gat786 @anushkrishnav can anyone please review this pr

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
We don't accept PRs without any corresponding Issue. It is difficult for us to find your issue in tons of issues available. We can't review and merged incomplete PR's.

**Comment by Bhumika2409** [comment]
Sorry @ankitaggarwal23 I didn't know that we have to link issue to the PR. This is for issue #6790 . So should I make a new issue?

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF
- Use proper display message how you are reading and writing outputs
- Add input-output prompts to make the program more interactive.

**Comment by Bhumika2409** [comment]
@ankitaggarwal23  could you please review the changes

File: C/operating_system/first_come_first_serve_algorithm.c [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+2. wt is waiting time .
+3. tat is turn around time (tat= wt+bt).
+4. avg_wt is average waiting time.
+5. avg_tat is average turn around time.
+6. total_wt is total waiting time.
+7. total_tat is total turn around time.
+8. process indiacates the index of process.
+9. n is the number of process.
+
+*/
```
Comment (ankitaggarwal23) [inline C/operating_system/first_come_first_serve_algorithm.c]: Rewrite it like that
```
/*

FCFS: First Come First Serve algorithm 

It is a CPU scheduling algorithm, in which the process are taken on the basis of their arrival time. 
The process that arrives first, is alloted the CPU first for the execution. Its working is similar to 
that of the FIFO in queue. It is a non-preemptive algorithm which means that the next process cannot be 
scheduled in the middle of the ongoing execution of a process in CPU.  

Terms explanation:
1. Burst time is the time taken by a process to complete its execution.
2. Waiting time is the total time for which a process has to wait for its chance to get the CPU for execution.
3. Turn around time is the total time between the submission of a process and its completion.

Implementation of the code:
Step 1: The number of process are scanned along with their respective burst time.
Step 2: Waiting time for a process is calculated by adding the burst time of the previous process. Waiting time for first process is taken as 0.
Step 3: Turn around time of a process is calculated which is the sum of burst time and waiting time of that process.
Step 4: Average waiting time and average turn around time are calculated by diving the total waiting and turn around time by nummber of process.

*/
```

File: C/operating_system/first_come_first_serve_algorithm.c [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+P[1]            4               0               4
+P[2]            8               4               12
+P[3]            6               12              18
+P[4]            3               18              21
+P[5]            7               21              28
+P[6]            2               28              30
+
+Average Waiting Time:13.833333
+Average Turnaround Time:18.833334
+
```
Comment (ankitaggarwal23) [inline C/operating_system/first_come_first_serve_algorithm.c]: Remove extra test case

File: C/operating_system/first_come_first_serve_algorithm.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+Complexities: Average case time complexity is of O(n^2) and the best case time complexity is of O(n). Space complexity is of O(1)
```
Comment (ankitaggarwal23) [inline C/operating_system/first_come_first_serve_algorithm.c]: Rewrite it like that
```
Complexities: 
Time complexity: O(n)
Space complexity: O(1)

```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Bhumika2409** [comment]
I'll do the changes

**Comment by Bhumika2409** [comment]
@ankitaggarwal23 can you please review the changes

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**